json.c

```diff
@@ -432,8 +432,10 @@ json_value * json_parse_ex (json_settings * settings,
 
                   case json_object:
 
-                     if (state.first_pass)
-                        (*(json_char **) &top->u.object.values) += string_length + 1;
+                     if (state.first_pass) {
+                        json_char **chars = (json_char **) &top->u.object.values;
```

Comment:
I actually didn't know C89 supported declaring variables at the beginning of any scope, I had always thought they had to be declared at the beginning of the function body, thanks for teaching me this new C89 factoid :)

Reply:
Honestly, I didn't know either.
I just wrote it not thinking about it and the compiler did not complain.
But yes, apparently the beginning of any scope is fine for variable declarations in C89.